### PR TITLE
fix(rule-finder): Support for unnamed scoped packages

### PR DESCRIPTION
**Pull request by bradzacher**
ESLint supports "unnamed" scoped plugins.
`@scope` translates to `@scope/eslint-plugin`.

This PR adds support for that case.

### CODE REVIEW

**Comment by sweetlikepete** [comment]
Hey Brad, this fix is awesome!

I found something that might be a problem. When you run this version of eslint-find-rules over an eslint config that contains all the rules for @getify/eslint-plugin-proper-arrows all rules come up as missing. 

I'm thinking this has something to do with this eslint-plugin using rule names in the format "@getify/proper-arrows/{rule-name}" instead of what every other plugin I've ever used does "proper-arrows/{ rule-name }".

Thanks again for updating this script. I hope this information is helpful.

**Comment by bradzacher** [comment]
Are you sure that's correct, @sweetlikepete?

Looking at the Readme for that plugin, it says you have to prefix the rule name with the scope

https://github.com/getify/eslint-plugin-proper-arrows/blob/master/README.md#eslintrcjson

**Comment by ljharb** [comment]
You definitely do; all scoped plugins and shareable configs work that way.

**Comment by sweetlikepete** [comment]
It's the only eslint plugin I have that is part of a scoped package and when I run eslint-find-rules all the rules it has are returned as unused in the format 'proper-arrows/params proper-arrows/name' when they added in my eslint rules config as '@getify/proper-arrows/params' and '@getify/proper-arrows/name'. The configuration is working correctly in eslint.

File: test/fixtures/eslint-with-deprecated-rules.json [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+    "@scope"
   ],
   "rules": {
     "foo-rule": [2],
     "old-rule": [2],
     "plugin/foo-rule": [2],
     "plugin/old-plugin-rule": [2],
     "scoped-plugin/foo-rule": [2],
-    "scoped-plugin/old-plugin-rule": [2]
+    "scoped-plugin/old-plugin-rule": [2],
```
Comment (ljharb) [inline test/fixtures/eslint-with-deprecated-rules.json]: this seems wrong to me; it should be `@scoped-plugin/`, not `scoped-plugin/`?

Comment (DylanVann) [reply test/fixtures/eslint-with-deprecated-rules.json]: That's what docs for `typescript-eslint` say - https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#usage

Comment (ljharb) [reply test/fixtures/eslint-with-deprecated-rules.json]: Those docs seem like they include the `@` in the rule name to me - are we looking at a different link?

Comment (bradzacher) [reply test/fixtures/eslint-with-deprecated-rules.json]: @DylanVann - typescript-eslint is a different case. it uses the special case name of `@typescript-eslint/eslint-plugin` which is automatically handled by eslint such that `/eslint-plugin` is not required anywhere.

However in this case, this looks like the old test cases were wrong.
Looking at an example of a package that doesn't use special naming (https://github.com/getify/eslint-plugin-proper-arrows#eslintrcjson), it should instead be `@scope/scoped-plugin/old-plugin-rule`

Comment (ljharb) [reply test/fixtures/eslint-with-deprecated-rules.json]: Right - the only thing that makes anything a scope is the leading @, ever

**Comment by bradzacher** [comment]
In regards to the failing tests - this is because `@scope` was only properly supported in ESLint v5.
Is it worth deleting the v4 test cases?
Or should we try and move them into a separate structure so v4 and v5 can be tested separately? (is it worth the effort considering v5.0.0 is almost a year old now?)

**Comment by ljharb** [comment]
Yes, it is still worth the effort to support as far back as we can.

Certainly if some functionality only is supported in eslint 5, we shouldn’t test it in < 5.

**Comment by bradzacher** [comment]
Ah small problem @ljharb.

By not testing scoped plugins/packages for `< v5`: the code coverage can never reach 100%, so the CI will always fail.
Off the top of my head, I don't think that there's any way to make this work without creating a file that's conditionally required?

Should I reconfigure travis to disable coverage checks for `< v4`?

**Comment by bradzacher** [comment]
I used an istanbul comment to specifically disable coverage on the if statement in question, as it cannot be tested on <5 due to the differing implementations.

Note that I removed `npm ls > /dev/null` from the CI script, as it broke on an unrelated dependency on the node4 test.

**Comment by ljharb** [comment]
Why does `npm ls` break on node 4? It shouldn't.

**Comment by bradzacher** [comment]
https://travis-ci.org/sarbbottam/eslint-find-rules/jobs/519370506#L851

One of the dependencies (har-validator) of dependencies (request) of dependencies (all-contributors-cli) uses a different AJV version and it causes it to error.

I think that the dependency resolution algorithm must be different in node v4?

What's the reason that it was running `npm ls`?

**Comment by ljharb** [comment]
In that case, please add `nvm install-latest-npm` to `before_install` in travis.yml, rather than removing `npm ls` (which, if it fails, means that nothing can be expected to work, because the dependency graph is invalid)

**Comment by bradzacher** [comment]
Done! That worked, so it must have been a change in the super old npm algorithm.
All tests are passing now!

File: .travis.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ node_js:
   - '6'
   - '4'
 before_script:
+  - nvm install-latest-npm
```
Comment (ljharb) [inline .travis.yml]: this needs to go in `before_install`, not `before_script`

Comment (bradzacher) [reply .travis.yml]: sorry!
I derped when I put it in and didn't read the suffix of the step....

**Comment by noamokman** [comment]
Any updates on this?

**Comment by noamokman** [comment]
@sarbbottam are you no longer maintaining this project?
Could you give permissions to other team members to advance this?
Thanks!

**Comment by ljharb** [comment]
@noamokman we're quite capable of merging it; it just needs a second review.

**Comment by ljharb** [comment]
to clarify: from a collaborator :-)

**Comment by noamokman** [comment]
@ljharb thanks for merging this!
Do you mind releasing a new version to npm?
Thanks again 😄

**Pull request merged**